### PR TITLE
A couple of fixes for 1592.

### PR DESCRIPTION
**Pull request by aseemr**
1592 partial:

- Adding support for names in `--using_facts_from`.
- `Modifies.Linear`.

### CODE REVIEW

File: src/smtencoding/FStar.SMTEncoding.Solver.fs [hunk 1]
```diff
@@ -106,43 +106,44 @@ let with_hints_db fname f =
...
-        | _ -> false
+        | Name lid -> Env.should_enc_lid e lid
+        | _ -> failwith "Solver.fs::filter_using_facts_from expected only Name fact ids"
     in
     let matches_fact_ids (include_assumption_names:BU.smap<bool>) (a:Term.assumption) =
       match a.assumption_fact_ids with
       | [] -> true //retaining `a` because it is not tagged with a fact id
       | _ ->
-        a.assumption_fact_ids |> BU.for_some should_enc_fid
+        a.assumption_fact_ids |> List.filter (function | Name _ -> true | _ -> false) |> BU.for_some should_enc_fid
```
Comment (nikswamy) [inline src/smtencoding/FStar.SMTEncoding.Solver.fs]: Why the double scan to filter first to Names and then to check should_enc_fid? Can't this be done in a single pass?

**Comment by nikswamy** [comment]
I noticed that the permission changed on Solver.fs from 644 to 755 ... can you change it back, please?

Also, left a small comment about the double scan when filitering facts.

Aside from that, it looks good. 

Thanks.

**Comment by aseemr** [comment]
Thanks! Will fix and merge.

**Comment by mtzguido** [comment]
Looks good to me too. Ideally we could clean up the history before merging so we have a single `snap` and avoid introducing 5cd9c2c and reverting it, but it's not critical.

**Comment by aseemr** [comment]
Thanks, cleaned up and merged.

**Pull request merged**